### DISCUSSION TITLE
this will cause an error in iOS clients

### DISCUSSION THREAD

**Discussion by notjulian** [bluenviron/mediamtx#3935] [Questions]
### Question

hi,
how can I fix this error?
2024/11/05 08:27:49 WAR [HLS] [muxer testrtsp] part duration changed from 280ms to 360ms - this will cause an error in iOS clients

thanks

**Comment by ptath** [comment]
metoo

```2025/01/01 08:24:15 WAR [HLS] [muxer cam6] part duration changed from 250ms to 267ms - this will cause an error in iOS clients```

maybe we should tune somehow this:

```
# Minimum duration of each part.
# A player usually puts 3 parts in a buffer before reproducing the stream.
# Parts are used in Low-Latency HLS in place of segments.
# Part duration is influenced by the distance between video/audio samples
# and is adjusted in order to produce segments with a similar duration.
hlsPartDuration: 200ms
```

?

**Reply by urnash** [reply]
2025/01/22 19:59:44 WAR [HLS] [muxer queue] part duration changed from 560ms to 550ms - this will cause an error in iOS clients

**Comment by notjulian** [comment]
I ended up using this https://github.com/AlexxIT/go2rtc

**Reply by urnash** [reply]
thank you, very good project, worked immediately without errors

**Reply by aler9** [reply]
Of course go2rtc does not show any warning (not errors) with respect to Low-latency HLS, it doesn't even provide it 😂

**Comment by twvacek** [comment]
I have this problem too for HEVC and LL-HLS. (Regular HLS seems to work fine---there are some log messages for resizing segment target duration, but it doesn't bring down some ios devices, but may explain some problems with others.) I'm trying to understand why segments and part target durations need resizing, and I'm not a proficient go programmer, either, so apologies in advance. It looks to me like the places (https://github.com/bluenviron/gohlslib/blob/715c3f8d868c058ec165bd2f8cc96094cb9e9043/muxer_stream.go#L746) where the issue is logged are simply trying to comply with the standard that does not allow the actual part or segment durations to be longer than the target. But why is the segmenter not respecting the target durations in the first place? The cameras (Amcrest) are configured to send key frames every 6 frames, and rame rate is 6 fps, so there should be plenty of these within a segment.

**Reply by twvacek** [reply]
In my case, the problem was caused by the incoming rtsp stream being proxied by gstreamer over a network, and the media timestamps were being replaced by their received time. Some setting changes preserved the original timestamps and fixed the issue. It looks like mediamtx had a latency buffer at one time...this might also be a avenue to resolve.
